src/bp/core/services/middleware/state-manager.ts

```diff
@@ -1,35 +1,86 @@
 import * as sdk from 'botpress/sdk'
+import { KnexExtension } from 'common/knex'
 import { BotpressConfig } from 'core/config/botpress.config'
 import { ConfigProvider } from 'core/config/config-loader'
+import Database from 'core/database'
 import { createExpiry } from 'core/misc/expiry'
-import { inject, injectable } from 'inversify'
+import { inject, injectable, tagged } from 'inversify'
+import { Redis } from 'ioredis'
+import Knex from 'knex'
 import _ from 'lodash'
 import { Memoize } from 'lodash-decorators'
+import ms from 'ms'
 
 import { SessionRepository, UserRepository } from '../../repositories'
 import { TYPES } from '../../types'
 import { SessionIdFactory } from '../dialog/session/id-factory'
+import { JobService } from '../job-service'
 import { KeyValueStore } from '../kvs'
 
+const getRedisSessionKey = sessionId => `userstate_${sessionId}`
```

Comment:
userstate can easily be confused with userAttributes is it ? if not can you change this for sessionstate_{sessionId}